Tribler/Core/APIImplementation/LaunchManyCore.py

```diff
@@ -594,31 +595,33 @@ def early_shutdown(self):
         """ Called as soon as Session shutdown is initiated. Used to start
         shutdown tasks that takes some time and that can run in parallel
         to checkpointing, etc.
+        :returns a Deferred that will fire once all dependencies acknowledge they have shutdown.
         """
         self._logger.info("tlm: early_shutdown")
 
         self.cancel_all_pending_tasks()
 
         # Note: sesslock not held
+        wait_defer_list = []
         self.shutdownstarttime = timemod.time()
         if self.torrent_checker:
-            self.torrent_checker.shutdown()
+            wait_defer_list.append(self.torrent_checker.shutdown())
```

Comment:
Hmmm ok, didn't know that. In my opinion those who are dependant on other shutdown's should wait for this in their own shutdown procedure. Do we change this with yielding for now? and inlinecallbacks?